src/pymor/algorithms/timestepping.py

```diff
@@ -10,221 +10,602 @@
+                dt_F = dt_F.assemble(mu)
+            def step_function(U_n, t_n):
+                t_np1 = t_n + dt
+                mu_t = mu.with_(t=t_np1)
+                return (
+                    M_dt_A.apply_inverse(M.apply(U_n, mu=mu_t) + dt_F.as_vector(mu_t), mu=mu_t, initial_guess=U_n),
+                    t_np1)
+
+        self.step_function = step_function
+
+    def _step_function(self, U_n, t_n):
+        return self.step_function(U_n, t_n)
+
 
 class ImplicitEulerTimeStepper(TimeStepper):
-    """Implict Euler time-stepper.
+    """Implicit Euler :class:`TimeStepper`.
 
     Solves equations of the form ::
 
-        M * d_t u + A(u, mu, t) = F(mu, t).
+        M * d_t u + A(u, mu, t) = F(mu, t),
+                     u(mu, t_0) = u_0(mu),
+
+    by an implicit Euler time integration, implemented in :class:`ImplicitEulerIterator`.
 
     Parameters
     ----------
     nt
         The number of time-steps the time-stepper will perform.
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each intermediate vector that is
+        calculated is returned. Else an interpolation of the calculated vectors on an equidistant temporal grid is
+        returned, using an appropriate interpolation of the respective time stepper.
     solver_options
-        The |solver_options| used to invert `M + dt*A`.
-        The special values `'mass'` and `'operator'` are
-        recognized, in which case the solver_options of
-        M (resp. A) are used.
+        The |solver_options| used to invert `M + dt*A`. The special values `'mass'` and `'operator'` are recognized,
+        in which case the solver_options of M (resp. A) are used.
     """
 
-    def __init__(self, nt, solver_options='operator'):
+    IteratorType = ImplicitEulerIterator
+
+    def __init__(self, nt, num_values=None, solver_options='operator', interpolation='P1'):
+        super().__init__(num_values, interpolation)
+
+        assert isinstance(nt, Number)
+        assert nt > 0
+
         self.__auto_init(locals())
 
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
-        return implicit_euler(operator, rhs, mass, initial_data, initial_time, end_time, self.nt, mu, num_values,
-                              solver_options=self.solver_options)
+
+class ExplicitEulerIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, iter, return_times)
+        self.__auto_init(locals())
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the step function U_np1 = M^{-1}(M U_n + dt F - dt A U_n)
+        if not isinstance(M, IdentityOperator) and (not M.parametric or 't' not in M.parameters):
```

Comment:
```suggestion
        if 't' not in M.parameters:
```
(Calling `assemble` on `IdentityOperator` won't hurt.)